package-res/resources/web/pentaho/type/complex.js

```diff
@@ -608,6 +608,46 @@ define([
       //endregion
       //endregion
 
+      //region serialization
+      serializeComplexValues: function(spec, values, keyArgs){
+        for (var property in values) {
```

Comment:
Check how the Complex#validate iterates over properties: https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/package-res/resources/web/pentaho/type/complex.js#L789